qiskit/qpy/__init__.py

```diff
@@ -400,10 +403,13 @@ def open(*args):
         char symbolic_encoding;
     }
 
+From V16 on, the file header struct is immediately followed by a circuit start table 
+containing the byte offsets of each circuit payload in the file (as an ``uint64_t``).
+
 All values use network byte order [#f1]_ (big endian) for cross platform
 compatibility.
 
-The file header is immediately followed by the circuit payloads.
+The circuit start table is immediately followed by the circuit payloads.
```

Comment:
I think it's important to be clear here that this is only the case prior to V16. From V16 on, there is no guarantee that we will store the payloads immediately after the circuit table nor sequentially for that matter 🙂.